From: Suparna Bhattacharya

wait_on_page_writeback_range shouldn't wait for pages beyond the
specified range. Ideally, the radix-tree-lookup could accept an
end_index parameter so that it doesn't return the extra pages
in the first place, but for now we just add a few extra checks
to skip such pages.


 filemap.c |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletion(-)

Index: linux.t/mm/filemap.c
===================================================================
--- linux.t.orig/mm/filemap.c	2004-08-13 10:33:10.303091760 -0400
diff -urNp linux-2.6.8/mm/filemap.c linux-2.6.8.SUSE/mm/filemap.c
--- linux-2.6.8/mm/filemap.c	2004-08-30 16:50:49.806347824 +0200
+++ linux-2.6.8.SUSE/mm/filemap.c	2004-08-30 16:51:25.053702149 +0200
@@ -230,6 +230,10 @@ static int wait_on_page_writeback_range(
 			if (page->index > end)
 				continue;
 
+			/* until radix tree lookup accepts end_index */
+			if (page->index > end) {
+				continue;
+			}
 			wait_on_page_writeback(page);
 			if (PageError(page))
 				ret = -EIO;
